text/0063-external-account-generator.md

```diff
@@ -0,0 +1,70 @@
+- FIP: 63
+- title: External Account Generator
+- author: Igor Shadurin, @IgorShadurin
+- status: draft
+- created: 2023-03-20
+
+# Summary
+An External Account Generator is a mechanism for the deterministic creation of FDP accounts based on the user's existing wallet without intermediate storage of the account in the data storage.
+
+# Context, motivation and guide level explanation
+
+Storing an FDP account means storing a mnemonic or seed in an open or encrypted form in any way: a sheet of paper, data storage, etc.
+
+This FIP describes how to create an FDP account with an external signer without storing the FDP account on third party storage and without exposing the external signer's private key or wallet address.
+
+An external signer is a secure Ethereum Wallet that does not expose the private key/mnemonic to third-party APIs and each data signature must be confirmed by the user. Metamask, Ledger, Trezor, etc. can act as such signers.
+
+On the one hand, such external signers provide the greatest security, but on the other hand, if they are used as a wallet for an FDP account, they create a huge flow of data signature confirmations from the user. Also, the mnemonic phrase of external signers should not be exposed to third party security applications, which makes it highly undesirable to use the mnemonic phrase as an FDP account. All these criteria call into question the possibility of using external signers as an FDP account in its current form.
+
+In order to meet the security and convenience criteria, two properties of the signed data of external signers can be used, namely, the determinism of the generated signature and the uniqueness of the external signer signature.
+
+Since no one but the owner of the external signer can create a valid signature for a particular data set, and sharing this signature does not affect security. This signature can be used as an entropy to initialize a mnemonic which can be used for initialization of an FDP account.
+
+There are several standards for signing data. Not every standard is supported by all external signers. One standard is supported at least by 3 popular (Metamask, Ledger, Trezor) external signers: `personal_sign` ([EIP-191](https://eips.ethereum.org/EIPS/eip-191)). This method also allows a person to see the data being signed before signing to make sure that the signature is used for the intended purpose.
+
+To generate a deterministic signature, a string must be passed to the signing function. When a user reads a string, it should be clear what the user is giving permission for and what will happen to the user data after the data is signed. The following text will be used as the signature string:
+
+`I am granting FULL ACCESS to the FDP account`

```

Comment:
But what if one has many external signers that user grants access to FDP account. 
I propose a "I am granting FULL access to the FDP account (0xfdpAccountAddress) from (0xExternalSignerAddress)"
This will differentiate external signers and will allow to link them to one fdp and vice versa.

Reply:
Right. My mistake of course you can not know the address before you have signature. Here you use signature as entropy for creating new mnemonic and from this signature ecrecover can also retrieve signer address. So you do not need anything but an arbitrary signed message. A signature acts as a password and should not be leaked.